Panic if first block range have no rollup events

#### System information

version: `v0.4.0' 

#### Expected behaviour
If first range of blocks have no rollup events must try next one

#### Actual behaviour
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1612f89]

goroutine 1 [running]:
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/l1_sync.(*L1SequentialSync).SyncBlocks(0xc00043c880, {0x1c33c48, 0xc00030bf40}, 0x0)
	/home/jesteban/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v0.3.4-0.20240527074759-5eb6f4a42ddd/synchronizer/l1_sync/l1_syncer_sequential.go:161 +0x689
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/internal.(*SynchronizerImpl).Sync(0xc0003971e0, 0x1)
	/home/jesteban/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v0.3.4-0.20240527074759-5eb6f4a42ddd/synchronizer/internal/synchronizer_impl.go:201 +0x5b8
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer.(*SynchronizerAdapter).Sync(0xc0003c0000, 0x1)
	/home/jesteban/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v0.3.4-0.20240527074759-5eb6f4a42ddd/synchronizer/synchronizer_adapter.go:31 +0x37
main.main()
	/home/jesteban/POLYGON/src/zkevm-sync-l1/client_meu/test-zkevm-sync-l1/main.go:66 +0x37c
```

#### Steps to reproduce the behaviour


#### Backtrace

````
[backtrace]
````

When submitting logs: please submit them as text and not screenshots.

--- a/synchronizer/l1_sync/l1_syncer_sequential.go
+++ b/synchronizer/l1_sync/l1_syncer_sequential.go
@@ -131,7 +131,7 @@ func (s *L1SequentialSync) SyncBlocks(ctx context.Context, lastEthBlockSynced *s
 	if lastEthBlockSynced != nil && lastEthBlockSynced.BlockNumber > 0 {
 		fromBlock = lastEthBlockSynced.BlockNumber
 	} else {
-		// fromBlock is the first block that we have, that the reson that we must sub 1 to genesis
+		// fromBlock contains the first block in the DB, therefore if we do not have any it is as if we had the one before genesis
 		fromBlock = s.cfg.GenesisBlockNumber - 1
 	}
 	blockRangeIterator := NewBlockRangeIterator(fromBlock, s.cfg.SyncChunkSize, blockPoints.L1LastBlockToSync)

--- a/synchronizer/l1_sync/l1_syncer_sequential.go
+++ b/synchronizer/l1_sync/l1_syncer_sequential.go
@@ -131,7 +131,8 @@ func (s *L1SequentialSync) SyncBlocks(ctx context.Context, lastEthBlockSynced *s
 	if lastEthBlockSynced != nil && lastEthBlockSynced.BlockNumber > 0 {
 		fromBlock = lastEthBlockSynced.BlockNumber
 	} else {
-		fromBlock = s.cfg.GenesisBlockNumber
+		// fromBlock is the first block that we have, that the reson that we must sub 1 to genesis
+		fromBlock = s.cfg.GenesisBlockNumber - 1
 	}
 	blockRangeIterator := NewBlockRangeIterator(fromBlock, s.cfg.SyncChunkSize, blockPoints.L1LastBlockToSync)
 
@@ -158,7 +159,11 @@ func (s *L1SequentialSync) SyncBlocks(ctx context.Context, lastEthBlockSynced *s
 		if blockRangeIterator.IsLastRange() {
 			break
 		}
-		blockRangeIterator = blockRangeIterator.NextRange(lastEthBlockSynced.BlockNumber)
+		if lastEthBlockSynced != nil {
+			blockRangeIterator = blockRangeIterator.NextRange(lastEthBlockSynced.BlockNumber)
+		} else {
+			blockRangeIterator = blockRangeIterator.NextRange(fromBlock)
+		}
 	}
 	return lastEthBlockSynced, true, nil
 }

--- a/synchronizer/l1_sync/l1_syncer_sequential_test.go
+++ b/synchronizer/l1_sync/l1_syncer_sequential_test.go
@@ -5,10 +5,12 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/0xPolygonHermez/zkevm-synchronizer-l1/etherman"
 	"github.com/0xPolygonHermez/zkevm-synchronizer-l1/state/entities"
 	mock_l1_check_block "github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/l1_check_block/mocks"
 	l1sync "github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/l1_sync"
 	mock_l1sync "github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/l1_sync/mocks"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/require"
 )
 
@@ -89,6 +91,22 @@ func TestSyncBlocksSequentialReorgDetected(t *testing.T) {
 	require.Error(t, err)
 }
 
+// First block to request is the genesis
+func TestSyncBlocksSequentialStartingNoBlock(t *testing.T) {
+	testData := newL1SyncData(t)
+	toBlock := uint64(140)
+	testData.mockBlockRetriever.EXPECT().GetL1BlockPoints(testData.ctx).Return(l1sync.BlockPoints{
+		L1LastBlockToSync:      toBlock,
+		L1FinalizedBlockNumber: toBlock,
+	}, nil)
+
+	testData.mockEth.EXPECT().GetRollupInfoByBlockRange(testData.ctx, uint64(testData.cfg.GenesisBlockNumber), &toBlock).Return(nil, nil, nil)
+	testData.mockBlockProcessor.EXPECT().ProcessBlockRange(testData.ctx, []etherman.Block(nil), map[common.Hash][]etherman.Order(nil), toBlock).Return(nil)
+
+	_, _, err := testData.sut.SyncBlocks(testData.ctx, nil)
+	require.NoError(t, err)
+}
+
 // --- HELPER FUNCTIONS ----------------------------------------------
 type testL1SyncData struct {
 	mockBlockRetriever *mock_l1sync.BlockPointsRetriever
@@ -99,6 +117,7 @@ type testL1SyncData struct {
 	sut                *l1sync.L1SequentialSync
 	ctx                context.Context
 	lastEthBlock       *entities.L1Block
+	cfg                l1sync.L1SequentialSyncConfig
 }
 
 func newL1SyncData(t *testing.T) *testL1SyncData {
@@ -107,10 +126,11 @@ func newL1SyncData(t *testing.T) *testL1SyncData {
 	mockState := mock_l1sync.NewStateL1SeqInterface(t)
 	mockBlockProcessor := mock_l1sync.NewBlockRangeProcessor(t)
 	mockReorg := mock_l1sync.NewReorgManager(t)
-	sut := l1sync.NewL1SequentialSync(mockBlock, mockEth, mockState, mockBlockProcessor, mockReorg, nil, l1sync.L1SequentialSyncConfig{
+	cfg := l1sync.L1SequentialSyncConfig{
 		SyncChunkSize:      100,
 		GenesisBlockNumber: 123,
-	})
+	}
+	sut := l1sync.NewL1SequentialSync(mockBlock, mockEth, mockState, mockBlockProcessor, mockReorg, nil, cfg)
 	ctx := context.TODO()
 	lastEthBlock := &entities.L1Block{
 		BlockNumber: 100,
@@ -125,5 +145,6 @@ func newL1SyncData(t *testing.T) *testL1SyncData {
 		sut:                sut,
 		ctx:                ctx,
 		lastEthBlock:       lastEthBlock,
+		cfg:                cfg,
 	}
 }

--- a/version.go
+++ b/version.go
@@ -7,7 +7,7 @@ import (
 )
 
 var (
-	Version = "v0.4.0"
+	Version = "v0.4.1"
 )
 
 // PrintVersion prints version info into the provided io.Writer.

